### ISSUE TITLE
[BUG] findShops count number, search results and offset limit

### ISSUE THREAD

**Issue by chyumin** [etsy/open-api#220]
**Describe the bug**
https://developers.etsy.com/documentation/reference#operation/findShops
- Response's `count` field is returning the number of results and not total count that matches the search term. The current count number is almost useless since we could just get the length of the results list if needed
- Search by `shop_name` are not concise. For example, by requesting with `shop_name` as "dong", some retrieved shops are: `TwoHworks`, `DGPTREASURES`, `TheBestLeathers` etc, which doesn't include the dong strings.
- `offset` has a (not documented) limit of 12400, if I do any request with offset = 12401 or bigger, I'd be redirected to the following page (not a json response):

![image](https://user-images.githubusercontent.com/12627814/133429822-dccdf8fc-97ba-43a5-8840-dca76a8101af.png)

**To Reproduce**
Do the following example requests and see the reported behaviour:
https://openapi.etsy.com/v3/application/shops?shop_name=dong&offset=0&limit=100
https://openapi.etsy.com/v3/application/shops?shop_name=dong&offset=0&limit=12401

**Expected behavior**
- `count` returns the total shops found by the search term
- results includes only shops that have search term in the shop_name
- `offset` with no limitations (or an json response with error and the limit documented)

**Additional context**
I'm trying to get a list of all shops, my strategy was searching by one char of 0-9 and a-z, and save all the results. The pointed bugs makes this strategy unsustainable. Please let me know if there is any better way to get data of all shops

**Comment by luiztauffer** [comment]
I agree, the search for all shops by name seems to be working incorrectly.

**Comment by chyumin** [comment]
The equivalent endpoint ([findAllShops](https://www.etsy.com/developers/documentation/reference/shop#method_findallshops)) for API v2 is documented: `Finds all Shops. If there is a keywords parameter, finds shops with shop_name starting with keywords.`

It'd be great to have such behavior in API v3

**Comment by chyumin** [comment]
@etsyachristensen Thanks for you answer and for fixing the total count
It'd be great to have the offset limit documented as well as a json error as response, instead of a text/html page
